### ISSUE TITLE
Lambda<Func<int>>(Block(Label(Label(typeof(int)), Constant(7)))) is not supported

### ISSUE THREAD

**Issue by dzmitry-lahoda** [dadhi/FastExpressionCompiler#78]
```
        [Test]
        [Ignore("Fails on IsNotNull")]
        public void BlockWithConstanRetrunIsSupported()
        {
            var ret = Block(Label(Label(typeof(int)), Constant(7)));
            var lambda = Lambda<Func<int>>(ret);
            var compiled = lambda.Compile();
            var value1 = compiled();
            Assert.AreEqual(7, value1);
            var fastCompiled = lambda.CompileFast<Func<int>>(true);
            Assert.IsNotNull(fastCompiled);
        }
```

**Comment by dzmitry-lahoda** [comment]
Trying to simplify (fails either)
```
        [Test]
        //[Ignore("Fails on IsNotNull")]
        public void ConstanRetrunIsSupported()
        { 
            var lambda = Lambda<Func<int>>(Label(Label(typeof(int)), Constant(7)));
            var compiled = lambda.Compile();
            var value1 = compiled();
            Assert.AreEqual(7, value1);
            var fastCompiled = lambda.CompileFast<Func<int>>(true);
            Assert.IsNotNull(fastCompiled);
        }
```

**Comment by dzmitry-lahoda** [comment]
With local variable retrurn to be sure nothing is broken
```
        [Test]
        //[Ignore("Fails on IsNotNull")]
        public void ConstanRetrunIsSupported()
        {
            var varr = Variable(typeof(int), "xxx");
            var assign = Assign(varr, Constant(7));
            var lambda = Lambda<Func<int>>(Block(new List<ParameterExpression> { varr }, assign, Label(Label(typeof(int)), varr)));
            var compiled = lambda.Compile();
            var value1 = compiled();
            Assert.AreEqual(7, value1);
            var fastCompiled = lambda.CompileFast<Func<int>>(true);
            Assert.IsNotNull(fastCompiled);
        }
```

**Comment by dzmitry-lahoda** [comment]
https://stackoverflow.com/questions/13687914/dynamic-linq-expression-with-return-value

int X() { return 7};

**Comment by jogibear9988** [comment]
Ist because of ExpressionType.Label seems to be not yet supported

**Comment by jogibear9988** [comment]
Goto is also missing

**Comment by jogibear9988** [comment]
with my pull req #103 your tests compile, but give wrong result

**Comment by jogibear9988** [comment]
fixed in #103
